modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,341 @@
+				$filtered_image = apply_filters( 'dominant_color_img_tag_add_adjust', $filtered_image, $context, $attachment_id );
+
+				if ( $filtered_image !== $match[0] ) {
+					$content = str_replace( $match[0], $filtered_image, $content );
+				}
+			}
+		}
+
+		return $content;
+	}
+
+	$filters = array( 'the_content', 'the_excerpt', 'widget_text_content', 'widget_block_content' );
+	foreach ( $filters as $filter ) {
+		add_filter( $filter, 'dominant_color_filter_content_tags', 20 );
+	}
+
+	add_filter( 'dominant_color_img_tag_add_adjust', 'dominant_color_tag_add_adjust', 10, 3 );
+}
+
+/**
+ * Add CSS needed for to show the dominant color as an image background.
+ */
+function dominant_color_add_inline_style() {
+	$handle = 'dominant-color-styles';
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = 'img[data-dominantcolor]:not(.has-transparency) { background-color: var(--dominant-color); background-clip: content-box, padding-box; }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_filter( 'wp_enqueue_scripts', 'dominant_color_add_inline_style' );
+
+
+/**
+ * Overloads wp_image_editors() to load the extended classes.
+ *
+ * @return string[]
+ */
+function dominant_color_set_image_editors() {
+
+	require_once 'class-dominant-color-image-editor-gd.php';
+	require_once 'class-dominant-color-image-editor-imagick.php';
+
+	return array( 'Dominant_Color_Image_Editor_GD', 'Dominant_Color_Image_Editor_Imagick' );
+}
+
+/**
+ * Get dominant color of image
+ *
+ * @param integer $id the image id.
+ * @param string  $default_color default color.
+ *
+ * @return string
+ */
+function dominant_color_get( $id, $default_color = 'eee' ) {
+
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	$file = get_attached_file( $id );
+
+	$dominant_color = wp_get_image_editor( $file )->get_dominant_color( $default_color );
+
+	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	return $dominant_color;
+}
+
+/**
+ * Works out if color has transparency
+ *
+ * @param integer $id the attachment id.
+ *
+ * @return bool
+ */
+function dominant_color_get_has_transparency( $id ) {
+
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	$file = get_attached_file( $id );
+
+	$editor = wp_get_image_editor( $file );
+	if ( is_wp_error( $editor ) ) {
+
+		return true; // safer to set to trans than not.
+	}
+	$has_transparency = wp_get_image_editor( $file )->get_has_transparency();
+
+	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	return $has_transparency;
+}
+
+/**
+ * Works out if the color is dark or light.
+ *
+ * @param string $hex color in hex.
+ *
+ * @return bool
+ */
+function dominant_color_color_is_light( $hex ) {
+	$hex       = str_replace( '#', '', $hex );
+	$r         = ( hexdec( substr( $hex, 0, 2 ) ) / 255 );
+	$g         = ( hexdec( substr( $hex, 2, 2 ) ) / 255 );
+	$b         = ( hexdec( substr( $hex, 4, 2 ) ) / 255 );
```

Comment:
Will this work for shorthand hex colors? Let's say `#ccc` instead of `#cccccc`?

Reply:
It didn't, good catch
I have added code to handle this along with tests